packages/dd-trace/src/config-helper.js

```diff
@@ -0,0 +1,90 @@
+'use strict'
+
+// TODO: Add jira tickets
+
+const { debuglog, deprecate } = require('util')
+const { supportedConfigurations, aliases, deprecations } = require('./supported-configurations.json')
+
+const aliasToCanonical = {}
+for (const canonical of Object.keys(aliases)) {
+  for (const alias of aliases[canonical]) {
+    if (aliasToCanonical[alias]) {
+      throw new Error(`The alias ${alias} is already used for ${aliasToCanonical[alias]}.`)
+    }
+    aliasToCanonical[alias] = canonical
+  }
+}
+
+// TODO: Consider to join deprecations with aliases by just making those entries an object.
+const deprecationMethods = {}
+for (const deprecation of Object.keys(deprecations)) {
+  deprecationMethods[deprecation] = deprecate(
+    () => {},
+    `The environment variable ${deprecation} is deprecated. Please use ${aliasToCanonical[deprecation]} instead.`,
+    `DATADOG_${deprecation}`
+  )
+}
+
+const debug = debuglog('dd:debug')
```

Comment:
Why use `debuglog` and not our own logger?

Reply:
It would cause a circular structure that is not possible to be resolved. We'd need the logger to be changed significantly.

Reply:
Are you sure? If I recall correctly, I removed the tight coupling between the config and logger in https://github.com/DataDog/dd-trace-js/pull/4515

Also, just tried this in the console which works:

<img width="497" alt="image" src="https://github.com/user-attachments/assets/48a0a7e6-423f-4e4e-ba1a-8b750f4f8940" />

Reply:
Ah sorry, I might have misunderstood. You specifically mention circular references, which I haven't checked. So I assume you've verified the circular reference hasn't been resolved by the time the logger is needed?

Reply:
Yes, that is the issue. I can have another look at it while it did not seem straight forward to resolve it last time I had a look.

Reply:
It would be prudent to make the logger work in situations like this. Having two different loggers is not ideal. That being said, the way the current logger works is great. So maybe it's not worth it trying to fix it. For example:
- You either enable all or nothing (I'd like to be able to enable logging just for a specific instrumentation or just for profiling for example)
- The environment variable to enable logging is called `DD_TRACE_DEBUG`... it's used outside of just traces and it's has a log level in the name, when in fact you you can change the level independently

In short, it's a bit of a mess.

Reply:
I removed the logging for now and added a TODO